sharktank/sharktank/models/llm/llm.py

```diff
@@ -196,16 +196,16 @@ def prefill(
                     count=len(self.cache.pipeline_to_device_map[pipeline]),
                     devices=self.cache.pipeline_to_device_map[pipeline],
                 )
-                for pipeline in range(self.cache.pipeline_count)
+                for pipeline in range(len(self.cache.pipeline_to_device_map))
```

Comment:
Why do we need this change?
What would happen if we are not using `PipelinedCache`? Do we know that we always have `pipeline_to_device_map` in the polymorphic KV cache? The other KV cache classes don't have this.
Maybe we should be getting this from the config.

Reply:
Also, I have to make this change because of the changes made here: https://github.com/nod-ai/shark-ai/pull/1404